### DISCUSSION TITLE
Is deta space suitable for quiz app that would have more than 1000+ records of questions data?

### DISCUSSION THREAD

**Discussion by tongsangcong** [deta/cloud-docs#223] [Help] [answered]
I have read the deta space starter doc on notion, it's amazing to get just focus on business rather than infra and tech, thank you Deta team

btw I just wonder, If one person get single deta base, so If the App is a quiz app where the questions itself is the same, everyone would have the same copy right? Bob will get the questions data same as alice does, it's alright for 10 or 50 records, but what about if we have a 1000 of record and 10+ users, is this case not suitable for deta space?

I mean, I want to update the questions record as much as possible, will it be a bad think for the infrastructure? because I don't even know how much questions I will store, It's much, I think would be 1000+, and in deta space, everyone will have the copy of all of that (it's copy by value right, not copy by reference)

**Comment by jnsougata** [comment]
You can assign separate deta base for individual users. And yes you can save a deep copy of the questions for each users to their respective deta base. Actually deta will make things easier in this case. Just go for it. As I have a discord bot with over 300K+ users and it uses Deta Base for storing user and server data from that perspective I can say deta can easily handle your requirements.

**Comment by abdelhai** [comment, marked as answer]
1. base can handle large amounts of data/traffic, even inside Space
2. data replication will rely on your implementation. you could have a central base that has the source of truth for the new questions, once new questions are published, the space apps will fetch them and store them in their "local" bases.
